test/end-to-end.js

```diff
@@ -970,6 +975,47 @@ describe('end-to-end', function() {
       });
     });
 
+    it('includes sensitive error details in development mode', function(done) {
+      var app = loopback({ localRegistry: true, loadBuiltinModels: true });
+      var bootOptions = {
+        appRootDir: SANDBOX,
+        env: 'development'
+      };
+      boot(app, bootOptions, function(err) {
+        if (err) return done(err);
+        app.listen(0, function() {
+          request(app)
+            .get('/url-does-not-exist')
+            .expect(404)
+            .end(function(err, res) {
+              var actualError = res.error;
+              expect(actualError).to.have.property('stack');
+              done();
+          });
+        });
+      });
+    });
+
+    it('omits sensitive error details in production mode', function(done) {
+      var app = loopback({ localRegistry: true, loadBuiltinModels: true });
+      var bootOptions = {
+        appRootDir: SANDBOX,
+        env: 'production'
+      };
+      boot(app, bootOptions, function(err) {
+        if (err) return done(err);
+        app.listen(0, function() {
+          request(app)
+            .get('/url-does-not-exist')
+            .expect(404)
+            .end(function(err, res) {
+              expect(JSON.stringify(res.body, null, 2)).to.not.contain(__filename)
```

Comment:
Please add a comment explaining what is the intent of this assertion, e.g.

```js
// Assert that the response body does not contain stack trace.
// We want the assertion to be robust and keep working even
// if the property name storing stack trace changes in the future,
// therefore we test full response body.
```

Also I just realised that `_filename` most likely won't be in the stack trace, because the error is created deep inside strong-remoting. Please check what is the actual `res.body` when `debug:true` and use a different string to detect stack trace. (In other words, make sure to see this test fail, to ensure the assertion is written in a way that can fail.)